### DISCUSSION TITLE
Personalized and customisable error messages

### DISCUSSION THREAD

**Discussion by Felienne** [hedyorg/hedy#3191] [Large Projects & Plans]
Currently, every wrong program leads to the same error message, but I think it would be really cool to add support for changing that, such that we can customize the error kids get with a faulty program.

We have done some explorations into different error messages with @TiBiBa's [thesis](https://github.com/Felienne/hedy/files/7419026/Master_Thesis__Gradual_Feedback_Model__GFM_FINALv2.pdf), see also this very old PR #1095. In the thesis/PR we were trying out "gradual" error messages, which change with each subsequent execution.

In addition to gradual error messages (which I still think is a great idea!) other use cases could be:

* giving kids the option to customize the error messages (do you want strict ones or kind ones? with or without suggestions?)
* allowing us to do another a/b test of various types of error messages (f.e. 1st "I did not understand that" versus 2nd person "you made a mistake" versus 3rd person "the computer did not understand you")

Of course this would impact both the front and the back end and would be a bit experimental, so it might be nice as a student project or a side quest from someone with free time and an exploratory nature!

**Comment by aspargillus** [comment]
For the public record: a team of students will start to work on this project on Monday, 19th Dec. 😃

**Reply by Felienne** [reply]
Fantastic! I have moved the discussion to the category of currently running student projects!
